crates/anthropic/src/anthropic.rs

```diff
@@ -293,13 +294,8 @@ pub async fn complete(
             .read_to_end(&mut body)
             .await
             .context("failed to read response body")?;
-        let body_str =
-            std::str::from_utf8(&body).context("failed to parse response body as UTF-8")?;
-        Err(AnthropicError::Other(anyhow!(
-            "Failed to connect to API: {} {}",
-            response.status(),
-            body_str
-        )))
+        let body_str = str::from_utf8(&body).context("failed to parse response body as UTF-8")?;
+        Err(AnthropicError::from_response(status.as_u16(), body_str))
```

Comment:
Today, `stream_completion` isn't attempting to parse the response body as JSON to look for structured errors (like `stream_completion_with_rate_limit_info` does; this `AnthropicError::from_response` centralizes the logic so they're both handling errors the same way.